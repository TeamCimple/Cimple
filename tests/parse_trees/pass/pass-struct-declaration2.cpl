struct Employee {
	int age;
	int salary;
};


int main(int argc){
	struct Employee *prof;

	prof.age = 1;
	prof.salary = 1000;

	return 0;
}
