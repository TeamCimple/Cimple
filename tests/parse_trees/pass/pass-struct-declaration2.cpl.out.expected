Struct((int Var(age),), (int Var(salary),), , Employee, Constructor(Body:
        CompoundStatement(Declarations:  StatementList: )),, )FuncDecl(Name: Var(main) ReturnType: int Parameters: PARAM(int Var(argc))  Body: CompoundStatement(Declarations: (Employee Pointer(Var(prof)),),  StatementList: Statement(EQUALS(Access(Var(Identifier(prof)),age), Int(1))), Statement(EQUALS(Access(Var(Identifier(prof)),salary), Int(1000))), RETURN(Int(0)))) 
